### PR TITLE
Implement user registration

### PR DESCRIPTION
**Pull request by ermanergoz**

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    suspend fun insertToDatabase(user: User)
+}
+
+class RegisterUserRepositoryImpl(
+    private val remoteDbUserDataSource: UserDataSource
+): RegisterUserRepository {
+    override suspend fun insertToDatabase(user: User) {
+        remoteDbUserDataSource.insertToDatabase(user)
+    }
+}
```
Comment (kgeriiie) [inline shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt:16]: I think you don't need an extra repository :) You can do all authentication-related stuff (login, register ) in authRepository. If you don't agree with me just ignore this comment :)

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/presentation/components/RegisterScreen.kt [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                        .asString(),
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text, imeAction = ImeAction.Next
+                    )
+                ) {
+                    onEvent(RegisterEvent.CityChanged(it))
+                }
+
+                UserInput(
+                    state = if (state.postalCode == DEFAULT_VALUE_UNFILLED) "" else state.postalCode.toString(),
```
Comment (kgeriiie) [inline shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/presentation/components/RegisterScreen.kt:240]: Hint: if you set postalCode as nullable you won't need this condition :)  
If it is a nullable variable you can use it like this:
state.postalCode?.toString().orEmpty() 

You don't have to modify it it is just a tip :)

File: shared/src/iosMain/kotlin/com/vml/tutorial/plantshop/core/utils/DateFormat.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        day = 1
+        second = (dateMillis / 1000)
+    }
+    val dateFormatter = NSDateFormatter().apply {
+        this.dateFormat = dateFormat
+        locale = NSLocale.currentLocale
+    }
+    val date = NSCalendar.currentCalendar.dateFromComponents(components)
+
+    return dateFormatter.stringFromDate(date ?: NSDate())
```
Comment (kgeriiie) [inline shared/src/iosMain/kotlin/com/vml/tutorial/plantshop/core/utils/DateFormat.kt:23]: Fun fact: yesterday I created a similar method :D nevermind you were faster :P but Can I show you a simple version of this?

        return NSDateFormatter().apply {
            dateFormat = format
            locale = NSLocale.currentLocale
        }.stringFromDate(NSDate.dateWithTimeIntervalSince1970(dateMillis.div(1000.0)))

**Review by kgeriiie — CHANGES_REQUESTED** [review]
Nice ;)

File: shared/src/iosMain/kotlin/com/vml/tutorial/plantshop/core/utils/DateFormat.kt (hunk 3 above)
Comment (ermanergoz) [reply shared/src/iosMain/kotlin/com/vml/tutorial/plantshop/core/utils/DateFormat.kt:23]: Thank you for sharing. I wasn't previously acquainted with the dateWithTimeIntervalSince1970 function. Refactored :)

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt (hunk 1 above)
Comment (ermanergoz) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt:16]: I plan to do more work on it, so I might remove it later if it's not needed

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/presentation/components/ProfileLettermark.kt [hunk 4]
```diff
@@ -30,7 +30,11 @@ fun ProfilePhoto(
             .background(MaterialTheme.colorScheme.secondaryContainer)
     ) {
         Text(
-            text = "${user?.firstName?.first()} ${user?.lastName?.first()}".uppercase(),
+            text = if (user?.firstName.isNullOrEmpty()) {
+                "${user?.email?.first()}"
+            } else {
+                "${user?.firstName?.first()} ${user?.lastName?.first()}".uppercase()
+            },
```
[the remark this answers is not in the document]
Comment (ermanergoz) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/presentation/components/ProfileLettermark.kt:37]: Sure

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt (hunk 1 above)
Comment (kgeriiie) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt:16]: Okay

File: shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/presentation/components/RegisterScreen.kt (hunk 2 above)
Comment (ermanergoz) [reply shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/presentation/components/RegisterScreen.kt]: Made it nullable

**Review by kgeriiie — APPROVED** [review]
Loved to read your code :) well done

**Pull request merged**